app/uk/gov/hmrc/helptosavestub/util/Delays.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavestub.util
+
+import akka.actor.Scheduler
+import akka.pattern.after
+import com.typesafe.config.Config
+import configs.syntax._
+import uk.gov.hmrc.helptosavestub.util.Delays.DelayConfig
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Random
+import scala.concurrent.duration._
+
+trait Delays {
+
+  val scheduler: Scheduler
+
+  implicit val ec: ExecutionContext
+
+  def withDelay[A](delayConfig: DelayConfig)(f: () ⇒ A): Future[A] =
+    if (delayConfig.enabled) {
+      after(nextDelay(delayConfig), scheduler)(Future(f()))
```

Comment:
during the delay is in action, what happens to the current thread thats handling the original request ?  will be be waiting or sleeping ?